internal/k8s/reconciler/route_test.go

```diff
@@ -266,6 +266,15 @@ func TestRouteMergedStatusAndBinding(t *testing.T) {
 	require.Equal(t, "expected", statuses[0].Conditions[0].Message)
 	require.Equal(t, RouteConditionReasonBindError, statuses[0].Conditions[0].Reason)
 
+	route.OnBindFailed(service.NewRefNotPermittedError("refnotpermitted expected"), gateway)
+
+	statuses = route.MergedStatus().Parents
+	require.Len(t, statuses, 3)
+	require.Equal(t, "expected", string(statuses[0].ParentRef.Name))
+	require.Equal(t, "expected", string(statuses[0].ControllerName))
+	require.Equal(t, "refnotpermitted expected", statuses[0].Conditions[0].Message)
+	require.Equal(t, RouteConditionReasonBindError, statuses[0].Conditions[0].Reason)
+
 	// check binding for non-existent route
 	gateway = NewK8sGateway(&gw.Gateway{
```

Comment:
So, I was thinking something more along the lines of this:

```suggestion
	// check route ref
	route = NewK8sRoute(inner, K8sRouteConfig{
		ControllerName: "expected",
		Logger:         hclog.NewNullLogger(),
	})
	route.resolutionErrors.Add(service.NewRefNotPermittedError("not found"))
	route.OnBindFailed(nil, gateway)
	statuses = route.MergedStatus().Parents
	require.Len(t, statuses, 3)
	require.Equal(t, "expected", string(statuses[0].ParentRef.Name))
	require.Equal(t, "expected", string(statuses[0].ControllerName))
	require.Equal(t, "not found", statuses[0].Conditions[1].Message)
	require.Equal(t, RouteConditionReasonRefNotPermitted, statuses[0].Conditions[1].Reason)

	// check binding for non-existent route
	route = NewK8sRoute(inner, K8sRouteConfig{
		ControllerName: "expected",
		Logger:         hclog.NewNullLogger(),
	})
	gateway = NewK8sGateway(&gw.Gateway{
```

Also, looks like you'll need to add some additional error concatenation code here:
https://github.com/hashicorp/consul-api-gateway/blob/a4fc8cd904df96e044563e57946783c818ee4424/internal/k8s/service/resolver.go#L120

since right now there's no casing for `refNotPermittedErrors`

Reply:
just to clarify, the above code didn't actually exercise the case/switch statement added to `OnBindFailed` since there wasn't anything in `resolutionErrors`